core/src/subject/SubjectEntity.ts

```diff
@@ -0,0 +1,250 @@
+import { PerspectiveProxy } from "../perspectives/PerspectiveProxy";
+import { makeRandomPrologAtom } from "./SDNADecorators";
+import { singularToPlural } from "./util";
+
+const flattenPrologList = (list: object): any[] => {
+  let result = []
+  while (list && list["head"]) {
+    result.push(list["head"])
+    list = list["tail"]
+  }
+  return result
+}
+
+export type QueryPartialEntity<T> = {
+  [P in keyof T]?: T[P] | (() => string);
+};
+
+
+export class SubjectEntity {
+  #baseExpression: string;
+  #subjectClass: string;
+  #perspective: PerspectiveProxy
+  author: string;
+  timestamp: string;
+
+  constructor(perspective: PerspectiveProxy, baseExpression?: string) {
+    this.#baseExpression = baseExpression ? baseExpression : `entry://${makeRandomPrologAtom(24)}`;
+    this.#perspective = perspective
+  }
+
+  get baseExpression() {
+    return this.#baseExpression
+  }
+
+  private async getData(id?: string) {
+    const tempId = id ?? this.#baseExpression;
+    let isInstance = await this.#perspective.isSubjectInstance(tempId, this.#subjectClass)
+    if (!isInstance) {
+      throw `Not a valid subject instance of ${this.#subjectClass} for ${tempId}`
+    }
+
+    const subject = await this.#perspective.getSubjectProxy(tempId, this.#subjectClass);
+
+    if (!subject) {
+      await this.#perspective.createSubject(this.#subjectClass, tempId);
+    }
+
+    let results = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property(C, Property)`)
+    let properties = results.map(result => result.Property)
+
+    for (let p of properties) {
+      const resolveExpressionURI = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property_resolve(C, "${p}")`)
+      const getProperty = async () => {
+        let results = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property_getter(C, "${tempId}", "${p}", Value)`)
+        if (results && results.length > 0) {
+          let expressionURI = results[0].Value
+
+          if (resolveExpressionURI) {
+            const expression = await this.#perspective.getExpression(expressionURI)
+            this['author'] = expression.author;
+            this['timestamp'] = expression.timestamp;
```

Comment:
That doesn't seem right. The expression we're setting in a property could be created by somebody else an definitely at a different timestamp than the base..